Model/Connection.php

```diff
@@ -119,20 +122,40 @@ public function getProfile()
         return $this->profile;
     }
 
+    public function getConnection()
+    {
+        return $this->connection;
+    }
+
     /**
      * Execute a command
      *
-     * @param string $command
-     * @return string
+     * @param $command
+     * @return bool|string
+     * @throws ConnectionException
+     *
      */
     public function exec($command)
     {
         $this->assertConnected();
 
-        $stream = ssh2_exec($this->connection, $command);
-        stream_set_blocking($stream, true);
-        $streamOut = ssh2_fetch_stream($stream, SSH2_STREAM_STDIO);
-        return stream_get_contents($streamOut);
+        if( !($stream = ssh2_exec($this->connection, $command)) ) {
+            throw new ConnectionException(
```

Comment:
ok to throw an exception if command fail, but not a ConnectionException. Maybe use deleted Exception, or create CommandException